pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -946,19 +974,182 @@ func TestReconcile_CustomTask(t *testing.T) {
 				t.Errorf("Expected reason %q but was %s", v1beta1.PipelineRunReasonRunning.String(), condition.Reason)
 			}
 
-			if len(reconciledRun.Status.Runs) != 1 {
-				t.Errorf("Expected PipelineRun status to include one Run status, got %d", len(reconciledRun.Status.Runs))
+			verifyRunStatusesCount(t, config.DefaultEmbeddedStatus, reconciledRun.Status, 1)
+			verifyRunStatusesNames(t, config.DefaultEmbeddedStatus, reconciledRun.Status, tc.wantRun.Name)
+		})
+	}
+}
+
+// TestReconcile_CustomTaskWithEmbeddedStatus runs "Reconcile" on a PipelineRun
+// with one Custom Task reference that has not been run yet, and each possible
+// value for the "embedded-status" feature flag. It verifies that the Run is
+// created, it checks the resulting API actions, status and events.
+func TestReconcile_CustomTaskWithEmbeddedStatus(t *testing.T) {
```

Comment:
can this follow the same pattern as Test_Reconcile above?

Reply:
Do you mean the separate function thingie? I didn’t bother splitting this one because it’s entirely new, so the diff would have been just as large either way. I guess there is value in consistency, though!

Reply:
yeah I think absent a strong reason to prefer either approach, let's keep things consistent

Reply:
👍